### PR TITLE
Criteo copy on write

### PR DESCRIPTION
**Pull request by jirayupiwon1**
/storage/emulated/0/Pictures/.thumbnails/.database_uuid

### CODE REVIEW

**Comment by uerceg** [comment]
Hey @Oklahoma1122,

This branch was just pushed for testing purposes and is not meant to be merged to `master` branch.

Cheers!

File: Adjust/ADJActivityPackage.h [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 #import "ADJActivityKind.h"
 
-@interface ADJActivityPackage : NSObject <NSCoding>
+@interface ADJActivityPackage : NSObject <NSCoding, NSCopying>
```
Comment (Paddy613) [inline Adjust/ADJActivityPackage.h:11]:
```suggestion
@interface ADJActivityPackage : NSObject <NSCoding, NSCopying>
```

File: Adjust/ADJActivityPackage.m [hunk 2]
```diff
@@ -88,4 +88,19 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
     [encoder encodeObject:self.partnerParameters forKey:@"partnerParameters"];
 }
 
+#pragma mark - NSCopying protocol methods
```
Comment (Paddy613) [inline Adjust/ADJActivityPackage.m:91]:
```suggestion
#pragma mark - NSCopying protocol methods
```

Comment (Paddy613) [reply Adjust/ADJActivityPackage.m:91]: ****

File: Adjust/ADJActivityPackage.m [hunk 3]
```diff
@@ -88,4 +88,19 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
...
+
+    copy.path = self.path;
+    copy.clientSdk = self.clientSdk;
+    copy.parameters = [self.parameters mutableCopyWithZone:zone];
+    copy.partnerParameters = [self.partnerParameters copyWithZone:zone];
+    copy.callbackParameters = [self.callbackParameters copyWithZone:zone];
+    copy.suffix = self.suffix;
+    copy.activityKind = self.activityKind;
+
+    return copy;
```
Comment (Paddy613) [inline Adjust/ADJActivityPackage.m:103]:
```suggestion
    return copy;
```

Comment (Paddy613) [reply Adjust/ADJActivityPackage.m:103]: ****

File: Adjust/ADJActivityPackage.m [hunk 4]
```diff
@@ -88,4 +88,19 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
...
+}
```
Comment (Paddy613) [inline Adjust/ADJActivityPackage.m:104]:
```suggestion
}
```

File: Adjust/ADJPackageHandler.m [hunk 5]
```diff
@@ -320,8 +320,11 @@ - (void)writePackageQueueS:(ADJPackageHandler *)selfS {
     if (selfS.packageQueue == nil) {
         return;
     }
-
```
Comment (Paddy613) [inline Adjust/ADJPackageHandler.m:323]:
```suggestion
    
```

Comment (Paddy613) [reply Adjust/ADJPackageHandler.m:323]: ****

File: Adjust/ADJPackageHandler.m [hunk 6]
```diff
@@ -320,8 +320,11 @@ - (void)writePackageQueueS:(ADJPackageHandler *)selfS {
     if (selfS.packageQueue == nil) {
         return;
     }
-    
-    [ADJUtil writeObject:selfS.packageQueue
+
```
Comment (Paddy613) [inline Adjust/ADJPackageHandler.m:323]:
```suggestion

```

File: Adjust/ADJPackageHandler.m [hunk 7]
```diff
@@ -320,8 +320,11 @@ - (void)writePackageQueueS:(ADJPackageHandler *)selfS {
     if (selfS.packageQueue == nil) {
         return;
     }
-    
-    [ADJUtil writeObject:selfS.packageQueue
```
Comment (Paddy613) [inline Adjust/ADJPackageHandler.m:324]: @jirayupiwon1

**Review by Paddy613 — COMMENTED** [review]
Hello-World

**Pull request closed without merging**